src/SearchNames.tsx

```diff
@@ -0,0 +1,64 @@
+import './MovieList.css'
+import type { JSONMovie } from './MovieList'
+import React from 'react'
+import './Header.css'
+
+interface Props {
+  names: JSONMovie[]
+}
+
+interface State {
+  searchName: string
+  filteredNames: JSONMovie[]
+}
+
+class SearchNames extends React.Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      searchName: '',
+      filteredNames: props.names
+    }
+  }
+
+  handleSearchNameChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const searchName = event.target.value
+    const filteredNames = this.props.names.filter(name =>
+      name.Title.toLowerCase().includes(searchName.toLowerCase())
+    )
+
+    localStorage.setItem('Input', searchName)
+
+    this.setState({ searchName, filteredNames })
+  }
+
```

Comment:
Input value should be saved to LocalStorage during the component’s unmount. During the initialization pick the value from LocalStorage and show it.

https://legacy.reactjs.org/docs/react-component.html#componentdidmount
https://legacy.reactjs.org/docs/react-component.html#componentwillunmount